lib/chief/command.rb

```diff
@@ -1,38 +1,42 @@
 module Chief
   class Command
     def self.call(*args, &block)
-      instance = new(*args, &block)
-      result = instance.call
-      if result.is_a?(Chief::Result) && instance.send(:created_result_object)
-        result
-      else
-        fail "#{method(__method__).receiver} must call success! or fail! and return the result"
-      end
+      new(*args, &block).call
     end
 
-    def self.value(*args)
-      call(*args).value
+    def self.value(*args, &block)
+      call(*args, &block).value
     end
 
-    def call
-      fail NotImplementedError, 'Implement #call in subclass'
-    end
+    def self.inherited(subclass)
+      subclass.send(:prepend, Module.new do
+        def call
+          obj = self
+          result = super
+          if result.is_a?(Chief::Result) && obj.send(:fullfilled)
+            result
+          else
+            fail "#{obj.class}#call must call success! or fail! and return the result"
+          end
+        end
 
-    def success!(value = true)
-      create_result_object(value, nil)
-    end
+        def success!(value = true)
+          create_result_object(value, nil)
+        end
 
-    def fail!(value = false, errors = true)
-      create_result_object(value, errors)
-    end
+        def fail!(value = false, errors = true)
+          create_result_object(value, errors)
```

Comment:
nothing to related to this PR, but just something to think about as u see more usages of chief.

i think the way we use failure objects is a bit confusing and can be simplified/improved

Reply:
I completely agree